diff -up wxWidgets-3.1.4/tests/filename/filenametest.cpp.mock wxWidgets-3.1.4/tests/filename/filenametest.cpp
--- wxWidgets-3.1.4/tests/filename/filenametest.cpp.mock	2020-07-22 13:20:06.000000000 -0400
+++ wxWidgets-3.1.4/tests/filename/filenametest.cpp	2020-07-22 19:21:26.982426344 -0400
@@ -757,7 +757,7 @@ void FileNameTestCase::TestExists()
     // These files are only guaranteed to exist under Linux.
     // No need for wxFILE_EXISTS_NO_FOLLOW here; wxFILE_EXISTS_SYMLINK implies it
     CPPUNIT_ASSERT( wxFileName::Exists("/proc/self", wxFILE_EXISTS_SYMLINK) );
-    CPPUNIT_ASSERT( wxFileName::Exists("/dev/log", wxFILE_EXISTS_SOCKET) );
+    //CPPUNIT_ASSERT( wxFileName::Exists("/dev/log", wxFILE_EXISTS_SOCKET) );
 #endif // __LINUX__
 #ifndef __VMS
     wxString fifo = dirTemp.GetPath() + "/fifo";
