From 957842946ea659a3ce4fb4a12aefb0e47c87a797 Mon Sep 17 00:00:00 2001
From: Scott Talbert <swt@techie.net>
Date: Thu, 23 Jul 2020 21:49:14 -0400
Subject: [PATCH] Skip tests for functions unsupported on WebKit2

---
 tests/controls/webtest.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/controls/webtest.cpp b/tests/controls/webtest.cpp
index 82b58597869..43a4073437f 100644
--- a/tests/controls/webtest.cpp
+++ b/tests/controls/webtest.cpp
@@ -123,6 +123,7 @@ TEST_CASE_METHOD(WebViewTestCase, "WebView", "[wxWebView]")
         CHECK(m_browser->CanGoForward());
     }
 
+#if !wxUSE_WEBVIEW_WEBKIT2
     SECTION("HistoryEnable")
     {
         LoadUrl();
@@ -136,7 +137,9 @@ TEST_CASE_METHOD(WebViewTestCase, "WebView", "[wxWebView]")
         CHECK(!m_browser->CanGoForward());
         CHECK(!m_browser->CanGoBack());
     }
+#endif
 
+#if !wxUSE_WEBVIEW_WEBKIT2
     SECTION("HistoryClear")
     {
         LoadUrl(2);
@@ -153,6 +156,7 @@ TEST_CASE_METHOD(WebViewTestCase, "WebView", "[wxWebView]")
         CHECK(!m_browser->CanGoForward());
         CHECK(!m_browser->CanGoBack());
     }
+#endif
 
     SECTION("HistoryList")
     {
